### DISCUSSION TITLE
Why it's not recommended to use env, build.env, builds in vercel.json?

### DISCUSSION THREAD

**Discussion by Josrph** [vercel/community#1453] [Help] [answered]
According to the [project config](https://vercel.com/docs/concepts/projects/project-configuration#env) page it's not recommended to use each of `env`, `build.env`, `builds`, and it always says "We recommend against using this property"???

**Comment by swarnava** [comment, marked as answer]
That's because pushing your environment variable directly to your codebase isn't a secure method. https://dev.to/somedood/please-dont-commit-env-3o9h

**Reply by Josrph** [reply]
OK, now I understand!!
